### PR TITLE
Populate platform attribute for Basic Station gateways 

### PR DESCRIPTION
**Pull request by KrishnaIyer**
#### Summary

Populate platform field for Basic Station gateways. Refs https://github.com/TheThingsIndustries/lorawan-stack/issues/2150

#### Changes

- Populate the `platform` attribute in the status message.
- Update tests.

#### Testing

Unit tests are sufficient since we're only adding values to existing messages

##### Regressions

Nope

#### Notes for Reviewers

Please also look at linked PR in `gatewaymapdata`.

#### Checklist

- [ ] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md`. The target branch is set to `master` if the changes are fully compatible with existing API, database, configuration and CLI.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/gatewayserver/io/basicstationlns/basicstationlns.go [hunk 1]
```diff
@@ -395,6 +395,7 @@ func (s *srv) handleTraffic(c echo.Context) (err error) {
 					"station":  version.Station,
 					"firmware": version.Firmware,
 					"package":  version.Package,
+					"platform": fmt.Sprintf("%s - Firmware %s - Protocol %d", version.Model, version.Firmware, version.Protocol),
```
Comment (johanstokking) [inline pkg/gatewayserver/io/basicstationlns/basicstationlns.go:398]: - Don't we need a separate `model` and `protocol`?
- Why is firmware in the metadata twice now?

Comment (KrishnaIyer) [reply pkg/gatewayserver/io/basicstationlns/basicstationlns.go:398]: > Don't we need a separate model and protocol?

Yeah but the `protocol` field is not used anywhere. If there's a need to export these metrics as part of events for example, I'll update it here based on what's needed.

> Why is firmware in the metadata twice now?

None of these fields are actually recognised by the NOC currently. It only knows the `platform` field. This is used [here](https://github.com/TheThingsIndustries/gatewaymapdata/pull/12) to extract the version.

Comment (johanstokking) [reply pkg/gatewayserver/io/basicstationlns/basicstationlns.go:398]: > None of these fields are actually recognised by the NOC currently. It only knows the `platform` field. This is used [here](https://github.com/TheThingsIndustries/gatewaymapdata/pull/12) to extract the version.

OK so we put a formatted string in here to wire through the V2 NOC, to regex it for the V2 gateway map. Please file a technical debt TODO issue and reference it here to remove this.

Comment (KrishnaIyer) [reply pkg/gatewayserver/io/basicstationlns/basicstationlns.go:398]: [Done](https://github.com/TheThingsNetwork/lorawan-stack/issues/2629)

Comment (johanstokking) [reply pkg/gatewayserver/io/basicstationlns/basicstationlns.go:398]: Missing the reference here

**Comment by KrishnaIyer** [comment]
@htdvisser: Can you take a final look here and the linked PR in `gatewaymapdata`?

**Pull request merged**